### PR TITLE
[ADD]hr_payroll_cancel from @vauxoo

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
## Cancel a payslip

This module introduces the following features:
- The module payroll_cancel allows the user to cancel a payslip whatever the previous state is.
### Installation

No special needs.
### Configuration

No extra configuration needed.
### Usage
1. Go to the payslip list and select the one you want to cancel.
2. Click on the button "Cancel payslip". The payslip is now rejected. 
3. To accept this payslip again, click on "set to draft" button.

If there’s a refund for a payslip the user should not cancel the entry because the refund would still be confirm. In that case, the user have either to confirm again the payslip or cancel the refund.

### CODE REVIEW

File: hr_payroll_cancel/test/update_payroll_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-
+  Workflow Mexican Electronic Payroll
+-
+  !python {model: hr.payslip}: |
```
Comment (moylop260) [inline hr_payroll_cancel/test/update_payroll_workflow.yml:4]: This should be a hook?

File: hr_payroll_cancel/hr_payslip_workflow.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        
+        <record id="workflow_transition_act_done_to_act_cancel" model="workflow.transition">
+            <field name="signal">cancel_sheet</field>
+            <field name="act_from" ref="hr_payroll.act_done"/>
+            <field name="act_to" ref="hr_payroll.act_cancel"/>
+            <field name="condition">True</field>
+        </record>
+        
+    </data>
+</openerp>
```
Comment (moylop260) [inline hr_payroll_cancel/hr_payslip_workflow.xml:20]: Missing newline

File: hr_payroll_cancel/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# -*- encoding: utf-8 -*-
+###########################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
```
Comment (moylop260) [inline hr_payroll_cancel/__openerp__.py:3]: Use short license format

File: hr_payroll_cancel/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Hr Payroll Cancel", 
+    "version": "1.0", 
+    "author": "Vauxoo", 
+    "category": "Localization/Mexico", 
+    "description": """
```
Comment (moylop260) [inline hr_payroll_cancel/__openerp__.py:31]: Add readme

File: hr_payroll_cancel/test/update_payroll_workflow.yml (hunk 1 above)
Comment (AaronHForgeFlow) [inline hr_payroll_cancel/test/update_payroll_workflow.yml:4]: What does a hook mean in this case?

Comment (moylop260) [inline hr_payroll_cancel/test/update_payroll_workflow.yml:4]: https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py

https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#installation-hooks

This os useful when you need a change after install the module

File: hr_payroll_cancel/i18n/es_MX.po [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Translation of OpenERP Server.
+# This file contains the translation of the following modules:
```
Comment (moylop260) [inline hr_payroll_cancel/i18n/es_MX.po:2]: .po files dont have a real translation
You could delete them and we can wait transifex

**Comment by moylop260** [comment]
Could you check the lint build, please?

https://travis-ci.org/OCA/hr/jobs/146074308

File: hr_payroll_cancel/test/update_payroll_workflow.yml (hunk 1 above)
Comment (AaronHForgeFlow) [inline hr_payroll_cancel/test/update_payroll_workflow.yml:4]: All right. I think this could be a post_init_hook. Do you agree?

Thank you for your answer.

Comment (moylop260) [inline hr_payroll_cancel/test/update_payroll_workflow.yml:4]: I'm agree with a `post_init_hook`

File: hr_payroll_cancel/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    "website": "http://www.vauxoo.com/", 
+    "license": "AGPL-3", 
+    "depends": [
+        "hr_payroll"
+    ], 
+    "demo": [], 
+    "data": [
+        "hr_payslip_view.xml", 
+        "hr_payslip_workflow.xml", 
+        "test/update_payroll_workflow.yml"
```
Comment (andhit-r) [inline hr_payroll_cancel/__openerp__.py:43]: It would be better to have python unittest

File: hr_payroll_cancel/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Hr Payroll Cancel", 
+    "version": "1.0", 
+    "author": "Vauxoo", 
+    "category": "Localization/Mexico",
```
Comment (andhit-r) [inline hr_payroll_cancel/__openerp__.py:30]: Localization?

File: hr_payroll_cancel/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Hr Payroll Cancel", 
+    "version": "1.0", 
+    "author": "Vauxoo",
```
Comment (andhit-r) [inline hr_payroll_cancel/__openerp__.py:29]: Add OCA as an author. See https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

File: hr_payroll_cancel/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Hr Payroll Cancel", 
+    "version": "1.0",
```
Comment (andhit-r) [inline hr_payroll_cancel/__openerp__.py:28]: Wrong version format. See https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

**Comment by andhit-r** [comment]
Hi @aheficent , 

Thanks for your contribution. I think you should preserve commit history from https://github.com/Vauxoo/addons-vauxoo where this module originated.

**Comment by AaronHForgeFlow** [comment]
Thank you for your answers. I agree with you, @andhit-r but I really don't know how to make a pull request to OCA from this fork: https://github.com/Eficent/addons-vauxoo/tree/8.0-hr_payroll_cancel

**Comment by moylop260** [comment]
You could use a similar process for extract modules history:
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#if-the-module-doesnt-exist-in-the-90-branch

**Comment by AaronHForgeFlow** [comment]
Thank you @moylop260 I am gonna do that and I will close this pull request.

**Comment by AaronHForgeFlow** [comment]
The new push request is here -> https://github.com/OCA/hr/pull/248

**Pull request closed without merging**